docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1519 @@
+
+typing.Union
++++++++++++++
+
+Some types comprise of multiple other types. For example, a ``MENTIONABLE`` type parameter can point to any of these:
+
+- :class:`discord.User`
+- :class:`discord.Member`
+- :class:`discord.Role`
+
+To specify in code, a parameter should annotate to a :obj:`typing.Union` with all the different models:
+
+.. code-block:: python
+
+    from typing import Union
+
+    @client.tree.command()
+    async def something(
+        interaction: discord.Interaction,
+        mentionable: Union[discord.User, discord.Member, discord.Role]
+    ):
+        await interaction.response.send_message(
+            f'i got: {mentionable}, of type: {mentionable.__class__.__name__}'
+        )
+
+Types that point to other types also don't have to include everything.
+For example, a ``CHANNEL`` type parameter can point to any channel in a guild,
+but can be narrowed down to a specific set of channels:
+
+.. code-block:: python
+
+    from typing import Union
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: discord.abc.GuildChannel):
+        # Everything except threads
+        pass
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: discord.TextChannel):
+        # Only text channels
+        pass
+
+    @client.tree.command()
+    async def channel_info(interaction: discord.Interaction, channel: Union[discord.Thread, discord.VoiceChannel]):
+        # Threads and voice channels only
+        pass
+
+.. note::
+
+    Union types can't mix Discord types.
+
+    Something like ``Union[discord.Member, discord.TextChannel]`` isn't possible.
+
+Refer to the :ref:`type conversion table <type_conversion>` for full information on sub-types.
+
+Describing
++++++++++++
+
+Descriptions are added to parameters using the :func:`.app_commands.describe` decorator,
+where each keyword is treated as a parameter name.
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.describe(
+        liquid="what type of liquid is on the wall",
+        amount="how much of it is on the wall"
+    )
+    async def bottles(interaction: discord.Interaction, liquid: str, amount: int):
+        await interaction.response.send_message(f'{amount} bottles of {liquid} on the wall!')
+
+These show up on Discord just beside the parameter's name:
+
+.. image:: /images/guide/app_commands/bottles_command_described.png
+
+Not specifying a description results with an ellipsis "..." being used instead.
+
+In addition to the decorator, parameter descriptions can also be added using
+Google, Sphinx or Numpy style docstrings.
+
+Examples using a command to add 2 numbers together:
+
+.. code-block:: python
+
+    @client.tree.command() # numpy
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        Parameters
+        -----------
+        a: int
```

Comment:
Pretty much the same exact thing as the last comment (command and param descriptions not being capitalized). This pretty much applies to the other two docstrings below.